src/gui/guiFormSpecMenu.cpp

```diff
@@ -3138,6 +3140,120 @@ void GUIFormSpecMenu::acceptInput(FormspecQuitMode quitmode=quit_mode_no)
 			fields["quit"] = "true";
 		}
 
+		#define BTOS(b) (b ? "true" : "false")
+
+		if (current_keys_pending.pending_mouse_event) {
+			SEvent::SMouseInput &event = current_keys_pending.mouse_event;
+
+			std::string field_type = "error";
+
+			#define SET_MOUSE_EVENT(e, t)	\
+				else if (event.Event == e)	\
+					field_type = t;
+
+			if (event.Event == EMIE_LMOUSE_PRESSED_DOWN)
```

Comment:
why doesn't this also use `SET_MOUSE_EVENT`? Is this meant to make it more obvious what it does?

Reply:
`SET_MOUSE_EVENT` uses the `else if` statement, whereas this first one needs the `if` statement, so it has to be different.